asmcomp/asmlink.ml

```diff
@@ -333,8 +334,9 @@ let call_linker file_list startup_file output_name =
     else if !Clflags.output_c_object then Ccomp.Partial
     else Ccomp.Exe
   in
-  if not (Ccomp.call_linker mode output_name files c_lib)
-  then raise(Error Linking_error)
+  let exitcode = Ccomp.call_linker mode output_name files c_lib in
+  if not (exitcode = 0)
+  then raise(Error(Linking_error (exitcode)))
```

Comment:
The parentheses around `exitcode` are unnecessary and using spacing for functoin calls is more idiomatic. I would reformulate this (and above) as

```
then raise (Error (Linking_error exitcode))
```